### PR TITLE
Yeoman generator for Artsy projects

### PR DESCRIPTION
**Pull request by orta**
yeoman

Takes the work from https://github.com/artsy/reaction-force and turns it into something that can be whipped up very quickly.

### CODE REVIEW

File: generators/app/index.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  setup() {
+    // For the license
+    var date = new Date();
+    this.year = date.getFullYear();
+
+    // Copy _all_ files, as templated, you can use ejx in them
+    var templateRoot = this.templatePath(this.type)
+    var projectRoot = this.destinationPath(this.name)
+
+    this.fs.copyTpl(templateRoot, projectRoot, this);
```
Comment (orta) [inline generators/app/index.js]: this is basically `cp template/**/** app` with ejs applied ( http://ejs.co )

File: generators/app/templates/web/scripts/sync-colors.sh [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+cp externals/elan/components/lib/variables/colors.json data
+
+# Let user know what's going on
+cd externals/elan
+META_SHA = $(git log -n1 --format='%H')
+echo "Elan color JSON is now at $($META_SHA)"
+
+# Save a commit
+cd ../..
+git add .
```
Comment (alloy) [inline generators/app/templates/web/scripts/sync-colors.sh]: This could commit code that is unrelated to sync and might not need committing at all. Do `git add data/colors.json` instead, or I think you can even move it into the `git commit` command.

File: generators/app/templates/web/scripts/sync-schema.sh [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+yarn install 
+yarn run dump-schema -- ../../data
+
+# Let user know what's going on
+META_SHA = $(git log -n1 --format='%H')
+echo "Metaphysics Schema is now at $($META_SHA)"
+
+# Save a commit
+cd ../..
+git add .
```
Comment (alloy) [inline generators/app/templates/web/scripts/sync-schema.sh]: Samesies.

**Comment by alloy** [comment]
🤘 

Giphy

**Comment by orta** [comment]
I don't know a one liner for commit stuff, but I added a rest to remove anything currently in staging, then just do the data folder

**Comment by xtina-starr** [comment]
Oooo, what's this new shiny nice looking thing?

File: generators/app/templates/web/scripts/sync-colors.sh [hunk 4]
```diff
@@ -11,7 +11,8 @@ echo "Elan color JSON is now at $($META_SHA)"
 
 # Save a commit
 cd ../..
-git add .
+git reset
+git add data/colors.json
```
Comment (alloy) [inline generators/app/templates/web/scripts/sync-colors.sh]: `git commit data/colors.json -m "Update Elan color JSON to $($META_SHA)"`

File: generators/app/templates/web/scripts/sync-schema.sh [hunk 5]
```diff
@@ -11,7 +11,8 @@ echo "Metaphysics Schema is now at $($META_SHA)"
 
 # Save a commit
 cd ../..
-git add .
+git reset
+git add data
```
Comment (alloy) [inline generators/app/templates/web/scripts/sync-schema.sh]: `git commit data/schema.* -m "Update Metaphysics schema to $($META_SHA)"`

**Comment by alloy** [comment]
I should’ve included examples the first time around, my bad.

**Comment by orta** [comment]
Alright - addressed that 👍

**Comment by alloy** [comment]
@xtina-starr A way for us to bootstrap a new project according to our up-to-date best practices in no time!
